src/Arcus.Messaging.Tests.Integration/Fixture/KeyRotationConfig.cs

```diff
@@ -10,24 +11,27 @@ public class KeyRotationConfig
         /// <summary>
         /// Initializes a new instance of the <see cref="KeyRotationConfig" /> class.
         /// </summary>
-        /// <param name="keyVaultSecret">The config to represent a Azure Key Vault secret.</param>
+        /// <param name="keyVault">The config to represent a Azure Key Vault secret.</param>
         /// <param name="servicePrincipal">The config to authenticate to Azure resources.</param>
-        /// <param name="serviceBusNamespace">The config to represent a Azure Service Bus Queue.</param>
-        public KeyRotationConfig(KeyVaultSecret keyVaultSecret, ServicePrincipal servicePrincipal, ServiceBusNamespace serviceBusNamespace)
+        /// <param name="serviceBusNamespace">The config to represent a Azure Service Bus namespace.</param>
+        /// <exception cref="ArgumentNullException">
+        ///     Thrown when the <paramref name="keyVault"/>, <paramref name="servicePrincipal"/>, or <paramref name="serviceBusNamespace"/> is <c>null</c>.
+        /// </exception>
+        public KeyRotationConfig(KeyVault keyVault, ServicePrincipal servicePrincipal, ServiceBusNamespace serviceBusNamespace)
         {
-            Guard.NotNull(keyVaultSecret, nameof(keyVaultSecret));
+            Guard.NotNull(keyVault, nameof(keyVault));
             Guard.NotNull(servicePrincipal, nameof(servicePrincipal));
             Guard.NotNull(serviceBusNamespace, nameof(serviceBusNamespace));
 
-            KeyVaultSecret = keyVaultSecret;
+            KeyVault = keyVault;
             ServicePrincipal = servicePrincipal;
             ServiceBusNamespace = serviceBusNamespace;
         }
 
         /// <summary>
         /// Gets the config representing a Azure Key Vault secret.
         /// </summary>
-        public KeyVaultSecret KeyVaultSecret { get; }
+        public KeyVault KeyVault { get; }
```

Comment:
Isn't this more clear?
```suggestion
        public KeyVaultConfig KeyVault { get; }
```